dev mode fails in failsafe verify goal when failsafe has systemPropertyVariables configured

Failed to run integration tests when failsafe plugin has systemPropertyVariables configured.
```
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-failsafe-plugin</artifactId>
                <version>${version.maven-failsafe-plugin}</version>
                <configuration>
                    <systemPropertyVariables>
                        <key>abc</key>
                    </systemPropertyVariables>
                </configuration>
            </plugin>
```

```
[INFO] Press the Enter key to run tests on demand.

[INFO] Running unit tests...
[INFO] Unit tests finished.
[INFO] Running integration tests...
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running it.io.openliberty.guides.rest.EndpointIT
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.928 s - in it.io.openliberty.guides.rest.EndpointIT
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[ERROR] Failed to run integration tests
[INFO] Press the Enter key to run tests on demand.
```
with debug:
```
[DEBUG] maven-failsafe-plugin configuration for verify phase: <?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <systemPropertyVariables>
    <key>abc</key>
    <liberty.hostname>192.168.0.21</liberty.hostname>
    <liberty.http.port>9080</liberty.http.port>
    <microshed_hostname>192.168.0.21</microshed_hostname>
    <microshed_http_port>9080</microshed_http_port>
    <wlp.user.dir>/Users/patricktiu/Projects/dev-ex/guide-rest-intro/finish/target/liberty/wlp/usr</wlp.user.dir>
  </systemPropertyVariables>
  <properties>
    <liberty.dev.test.run.id>1</liberty.dev.test.run.id>
  </properties>
  <additionalClasspathElements implementation="java.lang.String[]">${maven.test.additionalClasspath}</additionalClasspathElements>
  <argLine implementation="java.lang.String">${argLine}</argLine>
  <basedir default-value="${basedir}" implementation="java.io.File"/>
  <childDelegation default-value="false" implementation="boolean">${childDelegation}</childDelegation>
  <classpathDependencyExcludes implementation="java.lang.String[]">${maven.test.dependency.excludes}</classpathDependencyExcludes>
  <debugForkedProcess implementation="java.lang.String">${maven.failsafe.debug}</debugForkedProcess>
  <defaultClassesDirectory default-value="${project.build.outputDirectory}" implementation="java.io.File"/>
  <dependenciesToScan implementation="java.lang.String[]">${dependenciesToScan}</dependenciesToScan>
  <disableXmlReport default-value="false" implementation="boolean">${disableXmlReport}</disableXmlReport>
  <enableAssertions default-value="true" implementation="boolean">${enableAssertions}</enableAssertions>
  <encoding default-value="${project.reporting.outputEncoding}" implementation="java.lang.String">${encoding}</encoding>
  <excludedGroups implementation="java.lang.String">${excludedGroups}</excludedGroups>
  <excludesFile implementation="java.io.File">${failsafe.excludesFile}</excludesFile>
  <failIfNoSpecifiedTests implementation="java.lang.Boolean">${it.failIfNoSpecifiedTests}</failIfNoSpecifiedTests>
  <failIfNoTests implementation="java.lang.Boolean">${failIfNoTests}</failIfNoTests>
  <forkCount default-value="1" implementation="java.lang.String">${forkCount}</forkCount>
  <forkMode default-value="once" implementation="java.lang.String">${forkMode}</forkMode>
  <forkedProcessExitTimeoutInSeconds default-value="30" implementation="int">${failsafe.exitTimeout}</forkedProcessExitTimeoutInSeconds>
  <forkedProcessTimeoutInSeconds implementation="int">${failsafe.timeout}</forkedProcessTimeoutInSeconds>
  <groups implementation="java.lang.String">${groups}</groups>
  <includesFile implementation="java.io.File">${failsafe.includesFile}</includesFile>
  <junitArtifactName default-value="junit:junit" implementation="java.lang.String">${junitArtifactName}</junitArtifactName>
  <junitPlatformArtifactName default-value="org.junit.platform:junit-platform-engine" implementation="java.lang.String">${junitPlatformArtifactName}</junitPlatformArtifactName>
  <jvm implementation="java.lang.String">${jvm}</jvm>
  <localRepository default-value="${localRepository}" implementation="org.apache.maven.artifact.repository.ArtifactRepository"/>
  <objectFactory implementation="java.lang.String">${objectFactory}</objectFactory>
  <parallel implementation="java.lang.String">${parallel}</parallel>
  <parallelMavenExecution default-value="${session.parallel}" implementation="java.lang.Boolean"/>
  <parallelOptimized default-value="true" implementation="boolean">${parallelOptimized}</parallelOptimized>
  <parallelTestsTimeoutForcedInSeconds implementation="double">${failsafe.parallel.forcedTimeout}</parallelTestsTimeoutForcedInSeconds>
  <parallelTestsTimeoutInSeconds implementation="double">${failsafe.parallel.timeout}</parallelTestsTimeoutInSeconds>
  <perCoreThreadCount default-value="true" implementation="boolean">${perCoreThreadCount}</perCoreThreadCount>
  <pluginArtifactMap implementation="java.util.Map">${plugin.artifactMap}</pluginArtifactMap>
  <pluginDescriptor default-value="${plugin}" implementation="org.apache.maven.plugin.descriptor.PluginDescriptor"/>
  <printSummary default-value="true" implementation="boolean">${failsafe.printSummary}</printSummary>
  <projectArtifactMap implementation="java.util.Map">${project.artifactMap}</projectArtifactMap>
  <projectBuildDirectory default-value="${project.build.directory}" implementation="java.io.File"/>
  <redirectTestOutputToFile default-value="false" implementation="boolean">${maven.test.redirectTestOutputToFile}</redirectTestOutputToFile>
  <remoteRepositories default-value="${project.pluginArtifactRepositories}" implementation="java.util.List"/>
  <reportFormat default-value="brief" implementation="java.lang.String">${failsafe.reportFormat}</reportFormat>
  <reportNameSuffix default-value="" implementation="java.lang.String">${surefire.reportNameSuffix}</reportNameSuffix>
  <reportsDirectory default-value="${project.build.directory}/failsafe-reports" implementation="java.io.File"/>
  <rerunFailingTestsCount default-value="0" implementation="int">${failsafe.rerunFailingTestsCount}</rerunFailingTestsCount>
  <reuseForks default-value="true" implementation="boolean">${reuseForks}</reuseForks>
  <runOrder default-value="filesystem" implementation="java.lang.String">${failsafe.runOrder}</runOrder>
  <shutdown default-value="testset" implementation="java.lang.String">${failsafe.shutdown}</shutdown>
  <skip default-value="false" implementation="boolean">${maven.test.skip}</skip>
  <skipAfterFailureCount default-value="0" implementation="int">${failsafe.skipAfterFailureCount}</skipAfterFailureCount>
  <skipExec implementation="boolean">${maven.test.skip.exec}</skipExec>
  <skipITs implementation="boolean">${skipITs}</skipITs>
  <skipTests default-value="false" implementation="boolean">${skipTests}</skipTests>
  <suiteXmlFiles implementation="java.io.File[]">${failsafe.suiteXmlFiles}</suiteXmlFiles>
  <summaryFile default-value="${project.build.directory}/failsafe-reports/failsafe-summary.xml" implementation="java.io.File"/>
  <tempDir default-value="surefire" implementation="java.lang.String">${tempDir}</tempDir>
  <test implementation="java.lang.String">${it.test}</test>
  <testClassesDirectory default-value="${project.build.testOutputDirectory}" implementation="java.io.File"/>
  <testNGArtifactName default-value="org.testng:testng" implementation="java.lang.String">${testNGArtifactName}</testNGArtifactName>
  <testSourceDirectory default-value="${project.build.testSourceDirectory}" implementation="java.io.File"/>
  <threadCount implementation="int">${threadCount}</threadCount>
  <threadCountClasses default-value="0" implementation="int">${threadCountClasses}</threadCountClasses>
  <threadCountMethods default-value="0" implementation="int">${threadCountMethods}</threadCountMethods>
  <threadCountSuites default-value="0" implementation="int">${threadCountSuites}</threadCountSuites>
  <trimStackTrace default-value="true" implementation="boolean">${trimStackTrace}</trimStackTrace>
  <useFile default-value="true" implementation="boolean">${failsafe.useFile}</useFile>
  <useManifestOnlyJar default-value="true" implementation="boolean">${failsafe.useManifestOnlyJar}</useManifestOnlyJar>
  <useSystemClassLoader default-value="true" implementation="boolean">${failsafe.useSystemClassLoader}</useSystemClassLoader>
  <useUnlimitedThreads default-value="false" implementation="boolean">${useUnlimitedThreads}</useUnlimitedThreads>
  <workingDirectory implementation="java.io.File">${basedir}</workingDirectory>
  <project default-value="${project}" implementation="org.apache.maven.project.MavenProject"/>
  <session default-value="${session}" implementation="org.apache.maven.execution.MavenSession"/>
</configuration>
[DEBUG] Running executeMojo for Plugin [org.apache.maven.plugins:maven-failsafe-plugin]
[DEBUG] Attempting to load plugin Plugin [org.apache.maven.plugins:maven-failsafe-plugin] using pluginManager org.apache.maven.plugin.DefaultBuildPluginManager@5fa47fea and repositories [central (https://repo.maven.apache.org/maven2, default, releases)]
[DEBUG] Configuring mojo org.apache.maven.plugins:maven-failsafe-plugin:2.22.2:verify from plugin realm ClassRealm[plugin>org.apache.maven.plugins:maven-failsafe-plugin:2.22.2, parent: sun.misc.Launcher$AppClassLoader@55f96302]
[DEBUG] Configuring mojo 'org.apache.maven.plugins:maven-failsafe-plugin:2.22.2:verify' with basic configurator -->
[ERROR] Failed to run integration tests
[INFO] Press the Enter key to run tests on demand.
```

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java
@@ -30,11 +30,7 @@ import java.io.File;
 import java.io.FileReader;
 import java.io.IOException;
 import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
 import java.util.concurrent.ArrayBlockingQueue;
@@ -46,7 +42,6 @@ import org.apache.maven.execution.MavenSession;
 import org.apache.maven.model.Dependency;
 import org.apache.maven.model.Model;
 import org.apache.maven.model.Plugin;
-import org.apache.maven.model.PluginExecution;
 import org.apache.maven.model.Resource;
 import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
 import org.apache.maven.plugin.BuildPluginManager;
@@ -78,6 +73,7 @@ import io.openliberty.tools.common.plugins.util.PluginExecutionException;
 import io.openliberty.tools.common.plugins.util.PluginScenarioException;
 import io.openliberty.tools.common.plugins.util.ServerFeatureUtil;
 import io.openliberty.tools.common.plugins.util.ServerStatusUtil;
+import io.openliberty.tools.maven.utils.ExecuteMojoUtil;
 
 /**
  * Start a liberty server in dev mode import to set ResolutionScope for TEST as
@@ -618,18 +614,13 @@ public class DevMojo extends StartDebugMojoSupport {
         return updatedArtifacts;
     }
 
-    private void runTestMojo(String groupId, String artifactId, String phase) throws MojoExecutionException {
+    private void runTestMojo(String groupId, String artifactId, String goal) throws MojoExecutionException {
         Plugin plugin = getPlugin(groupId, artifactId);
-        Xpp3Dom config = getPluginConfig(plugin, phase);
-        if (config == null) {
-            log.debug("Could not find " + artifactId + " configuration for " + phase
-                    + " phase. Creating new configuration.");
-            config = configuration();
-        }
+        Xpp3Dom config = ExecuteMojoUtil.getPluginGoalConfig(plugin, goal, log);
 
-        if (phase.equals("test")) {
+        if (goal.equals("test")) {
             injectTestId(config);
-        } else if (phase.equals("integration-test")) {
+        } else if (goal.equals("integration-test")) {
             injectTestId(config);
             injectLibertyProperties(config);
             // clean up previous summary file
@@ -651,9 +642,9 @@ public class DevMojo extends StartDebugMojoSupport {
             } else {
                 log.debug("Summary file doesn't exist");
             }
-        } else if (phase.equals("failsafe-report-only")) {
+        } else if (goal.equals("failsafe-report-only")) {
             Plugin failsafePlugin = getPlugin("org.apache.maven.plugins", "maven-failsafe-plugin");
-            Xpp3Dom failsafeConfig = getPluginConfig(failsafePlugin, "integration-test");
+            Xpp3Dom failsafeConfig = ExecuteMojoUtil.getPluginGoalConfig(failsafePlugin, "integration-test", log);
             Xpp3Dom linkXRef = new Xpp3Dom("linkXRef");
             if (failsafeConfig != null) {
                 Xpp3Dom reportsDirectoryElement = failsafeConfig.getChild("reportsDirectory");
@@ -669,9 +660,9 @@ public class DevMojo extends StartDebugMojoSupport {
             }
             linkXRef.setValue("false");
             config.addChild(linkXRef);
-        } else if (phase.equals("report-only")) {
+        } else if (goal.equals("report-only")) {
             Plugin surefirePlugin = getPlugin("org.apache.maven.plugins", "maven-surefire-plugin");
-            Xpp3Dom surefireConfig = getPluginConfig(surefirePlugin, "test");
+            Xpp3Dom surefireConfig = ExecuteMojoUtil.getPluginGoalConfig(surefirePlugin, "test", log);
             Xpp3Dom linkXRef = new Xpp3Dom("linkXRef");
             if (surefireConfig != null) {
                 Xpp3Dom reportsDirectoryElement = surefireConfig.getChild("reportsDirectory");
@@ -688,9 +679,9 @@ public class DevMojo extends StartDebugMojoSupport {
             linkXRef.setValue("false");
             config.addChild(linkXRef);
         }
-        log.debug(artifactId + " configuration for " + phase + " phase: " + config);
 
-        executeMojo(plugin, goal(phase), config, executionEnvironment(project, session.clone(), pluginManager));
+        log.debug(groupId + ":" + artifactId + " " + goal + " configuration:\n" + config);
+        executeMojo(plugin, goal(goal), config, executionEnvironment(project, session.clone(), pluginManager));
     }
 
     /**
@@ -706,41 +697,6 @@ public class DevMojo extends StartDebugMojoSupport {
             plugin = plugin(groupId(groupId), artifactId(artifactId), version("RELEASE"));
         }
         return plugin;
-    }
-
-    /**
-     * Given the Plugin get the Xpp3Dom configuration
-     * 
-     * @param plugin
-     * @param phase
-     * @return configuration specified in pom.xml
-     */
-    private Xpp3Dom getPluginConfig(Plugin plugin, String phase) {
-        Xpp3Dom config = null;
-
-        List<Plugin> buildPlugins = project.getBuildPlugins();
-        for (Plugin p : buildPlugins) {
-            if (p.equals(plugin)) {
-                config = (Xpp3Dom) p.getConfiguration();
-
-                PluginExecution pluginExecution = null;
-                Map<String, PluginExecution> peMap = p.getExecutionsAsMap();
-
-                String[] defaultExecutionIds = new String[] { "default-" + phase, phase, "default" };
-                for (String executionId : defaultExecutionIds) {
-                    pluginExecution = peMap.get(executionId);
-                    if (pluginExecution != null) {
-                        break;
-                    }
-                }
-                if (pluginExecution != null) {
-                    Xpp3Dom executionConfig = (Xpp3Dom) pluginExecution.getConfiguration();
-                    config = Xpp3Dom.mergeXpp3Dom(executionConfig, config);
-                }
-                break;
-            }
-        }
-        return config;
     }
 
     /**
@@ -914,153 +870,41 @@ public class DevMojo extends StartDebugMojoSupport {
         return plugin;
     }
 
-    private Xpp3Dom getLibertyPluginConfig() {
-        // get the Liberty plugin configuration from the pom and overrides looseApplication to true.
-        // ignores the plugin execution configuration just like CLI invocation (e.g. mvn liberty:create) ignoring execution configuration.
-        Plugin libertyPlugin = getLibertyPlugin();
-        Xpp3Dom overrides = configuration(element(name("looseApplication"), "true"));
-        return getPluginConfig(libertyPlugin, overrides);
-    }
-
-    private Xpp3Dom getPluginConfig(Plugin plugin, Xpp3Dom overrides) {
-        Xpp3Dom config = (overrides != null) ? overrides : configuration();
-        Xpp3Dom pluginConfig = (Xpp3Dom)plugin.getConfiguration();
-        if (pluginConfig != null) {
-            config = Xpp3Dom.mergeXpp3Dom(config, pluginConfig);
-        }
-        return config;
-    }
-
-    private static final ArrayList<String> commonParams = new ArrayList<>(Arrays.asList(
-            "installDirectory", "assemblyArchive", "assemblyArtifact", "libertyRuntimeVersion",
-            "install", "licenseArtifact", "serverName", "userDirectory", "outputDirectory",
-            "assemblyInstallDirectory", "refresh", "skip"
-            // executeMojo can not use alias parameters:
-            // "runtimeArchive", "runtimeArtifact", "runtimeInstallDirectory"
-            ));
-    
-    private static final ArrayList<String> commonServerParams = new ArrayList<>(Arrays.asList(
-            "serverXmlFile", "configDirectory", "bootstrapProperties", "bootstrapPropertiesFile",
-            "jvmOptions", "jvmOptionsFile", "serverEnvFile"
-            // executeMojo can not use alias parameters:
-            // "configFile", "serverEnv"
-            ));
-    
-    private static ArrayList<String> createParams;
-    static {
-        createParams = new ArrayList<>(Arrays.asList(
-                "template", "libertySettingsFolder", "noPassword"
-                ));
-        createParams.addAll(commonParams);
-        createParams.addAll(commonServerParams);
-    }
-    
-    private static ArrayList<String> deployParams;
-    static {
-        deployParams = new ArrayList<>(Arrays.asList(
-                "appsDirectory", "stripVersion", "deployPackages", "looseApplication", "timeout"
-                // executeMojo can not use alias parameters:
-                // "installAppPackages"
-                ));
-        deployParams.addAll(commonParams);
-        deployParams.addAll(commonServerParams);
-    }
-    
-    private static ArrayList<String> installFeatureParams;
-    static {
-        installFeatureParams = new ArrayList<>(Arrays.asList("features"));
-        installFeatureParams.addAll(commonParams);
-    }
-
-    private static final Map<String, String> aliasMap;
-    static {
-        Map<String, String>tempMap = new HashMap<String, String>();
-        tempMap.put("runtimeArtifact", "assemblyArtifact");
-        tempMap.put("runtimeArchive", "assemblyArchive");
-        tempMap.put("runtimeInstallDirectory", "assemblyInstallDirectory");
-        tempMap.put("configFile", "serverXmlFile");
-        tempMap.put("serverEnv", "serverEnvFile");
-        tempMap.put("installAppPackages", "deployPackages");
-        aliasMap = Collections.unmodifiableMap(tempMap);
-    }
-
-    private Xpp3Dom stripConfigElements(Xpp3Dom config, ArrayList<String> goalParams) {
-        // convert alias parameter key to actual parameter key
-        Xpp3Dom alias;
-        for (String key : aliasMap.keySet()) {
-            alias = config.getChild(key);
-            if (alias != null) {
-                if ("runtimeArtifact".contentEquals(key)) {
-                    Xpp3Dom artifact = new Xpp3Dom(aliasMap.get(key));
-                    for (Xpp3Dom child : alias.getChildren()) {
-                        artifact.addChild(child);
-                    }
-                    config.addChild(artifact);
-                } else {
-                    Element e = (element(name(aliasMap.get(key)), alias.getValue()));
-                    config.addChild(e.toDom());
-                }
-            }
-        }
-
-        // strip non applicable parameters
-        List<Integer> removeChildren = new ArrayList<Integer>();
-        for (int i=0; i<config.getChildCount(); i++) {
-            if (!goalParams.contains(config.getChild(i).getName().trim())) {
-                removeChildren.add(i);
-            }
-        }
-        Collections.reverse(removeChildren);
-        for (int child : removeChildren) {
-            config.removeChild(child);
-        }
-        return config;
-    }
-
     private void runLibertyMojoCreate() throws MojoExecutionException {
-        Xpp3Dom config = stripConfigElements(getLibertyPluginConfig(), createParams);
-        log.info("Running liberty:create goal");
+        Xpp3Dom config = ExecuteMojoUtil.getPluginGoalConfig(getLibertyPlugin(), "create", log);
         runLibertyMojo("create", config);
     }
 
     private void runLibertyMojoDeploy() throws MojoExecutionException {
-        Xpp3Dom pluginConfig = (Xpp3Dom)getLibertyPlugin().getConfiguration();
-        if (pluginConfig != null && pluginConfig.getChild("looseApplication") != null 
-                && "false".equals(pluginConfig.getChild("looseApplication").getValue())) {
+        Xpp3Dom config = ExecuteMojoUtil.getPluginGoalConfig(getLibertyPlugin(), "deploy", log);
+        Xpp3Dom looseApp = config.getChild("looseApplication");
+        if (looseApp != null && "false".equals(looseApp.getValue())) {
             log.warn("Overriding liberty plugin pararmeter, \"looseApplication\" to \"true\" and deploying application in looseApplication format");
+            looseApp.setValue("true");
         }
-        Xpp3Dom config = stripConfigElements(getLibertyPluginConfig(), deployParams);
-        log.info("Running liberty:deploy goal");
         runLibertyMojo("deploy", config);
     }
 
     private void runLibertyMojoInstallFeature(Element features) throws MojoExecutionException {
-        Xpp3Dom config;
-
-        if (features == null) {
-            config = stripConfigElements(getLibertyPluginConfig(), installFeatureParams);
-        } else {
-            config = Xpp3Dom.mergeXpp3Dom(configuration(features), stripConfigElements(getLibertyPluginConfig(), installFeatureParams));
+        Xpp3Dom config = ExecuteMojoUtil.getPluginGoalConfig(getLibertyPlugin(), "install-feature", log);;
+        if (features != null) {
+            config = Xpp3Dom.mergeXpp3Dom(configuration(features), config);
         }
-
-        log.info("Running liberty:install-feature goal");
         runLibertyMojo("install-feature", config);
     }
 
     private void runLibertyMojo(String goal, Xpp3Dom config) throws MojoExecutionException {
-        log.debug("LibertyMojo:" + goal + " configuration:\n" + config);
+        log.info("Running liberty:" + goal);
+        log.debug("configuration:\n" + config);
         executeMojo(getLibertyPlugin(), goal(goal), config,
                 executionEnvironment(project, session, pluginManager));
     }
 
     private void runMojo(String groupId, String artifactId, String goal) throws MojoExecutionException {
         Plugin plugin = getPlugin(groupId, artifactId);
-        Xpp3Dom config = (Xpp3Dom)plugin.getConfiguration();
-        if (config == null) {
-            config = configuration();
-        }
+        Xpp3Dom config = ExecuteMojoUtil.getPluginGoalConfig(plugin, goal, log);
         log.info("Running " + artifactId + ":" + goal);
-        log.debug(groupId + ":" + artifactId + " " + goal + " configuration:\n" + config);
+        log.debug("configuration:\n" + config);
         executeMojo(plugin, goal(goal), config,
                 executionEnvironment(project, session, pluginManager));
     }

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/utils/ExecuteMojoUtil.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/utils/ExecuteMojoUtil.java
@@ -20,7 +20,7 @@ import org.twdata.maven.mojoexecutor.MojoExecutor.Element;
 public class ExecuteMojoUtil {
 
     // https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html
-    private static final ArrayList<String> compileParams = new ArrayList<>(Arrays.asList(
+    private static final ArrayList<String> COMPILE_PARAMS = new ArrayList<>(Arrays.asList(
             "annotationProcessorPaths", "annotationProcessors", "compilerArgs", "compilerArgument",
             "compilerArguments", "compilerId", "compilerReuseStrategy", "compilerVersion", "debug",
             "debuglevel", "encoding", "excludes", "executable", "failOnError", "failOnWarning",
@@ -32,7 +32,7 @@ public class ExecuteMojoUtil {
     ));
 
     // https://maven.apache.org/plugins/maven-compiler-plugin/testCompile-mojo.html
-    private static final ArrayList<String> testCompileParams = new ArrayList<>(Arrays.asList(
+    private static final ArrayList<String> TEST_COMPILE_PARAMS = new ArrayList<>(Arrays.asList(
             "annotationProcessorPaths", "annotationProcessors", "compilerArgs", "compilerArgument",
             "compilerArguments", "compilerId", "compilerReuseStrategy", "compilerVersion", "debug",
             "debuglevel", "encoding", "executable", "failOnError", "failOnWarning", 
@@ -45,7 +45,7 @@ public class ExecuteMojoUtil {
     ));
 
     // https://maven.apache.org/plugins/maven-resources-plugin/resources-mojo.html
-    private static final ArrayList<String> resourcesParams = new ArrayList<>(Arrays.asList(
+    private static final ArrayList<String> RESOURCES_PARAMS = new ArrayList<>(Arrays.asList(
             "outputDirectory", "addDefaultExcludes", "delimiters", "encoding", "escapeString",
             "escapeWindowsPaths", "fileNameFiltering", "filters", "includeEmptyDirs", 
             "mavenFilteringHints", "nonFilteredFileExtensions", "overwrite", "skip",
@@ -53,7 +53,7 @@ public class ExecuteMojoUtil {
     ));
 
     // https://maven.apache.org/plugins/maven-resources-plugin/testResources-mojo.html
-    private static final ArrayList<String> testResourcesParams = new ArrayList<>(Arrays.asList(
+    private static final ArrayList<String> TEST_RESOURCES_PARAMS = new ArrayList<>(Arrays.asList(
             "outputDirectory", "resources", "addDefaultExcludes", "delimiters", "encoding",
             "escapeString", "escapeWindowsPaths", "fileNameFiltering", "filters",
             "includeEmptyDirs", "mavenFilteringHints", "nonFilteredFileExtensions",
@@ -62,7 +62,7 @@ public class ExecuteMojoUtil {
     ));
 
     // https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html
-    private static final ArrayList<String> testParams = new ArrayList<>(Arrays.asList(
+    private static final ArrayList<String> TEST_PARAMS = new ArrayList<>(Arrays.asList(
             "testSourceDirectory", "additionalClasspathElements", "argLine", "basedir",
             "childDelegation", "classesDirectory", "classpathDependencyExcludes", 
             "classpathDependencyScopeExclude", "debugForkedProcess", "dependenciesToScan",
@@ -85,7 +85,7 @@ public class ExecuteMojoUtil {
     ));
 
     // https://maven.apache.org/surefire/maven-failsafe-plugin/integration-test-mojo.html
-    private static final ArrayList<String> integrationTestParams = new ArrayList<>(Arrays.asList(
+    private static final ArrayList<String> INTEGRATION_TEST_PARAMS = new ArrayList<>(Arrays.asList(
             "summaryFile", "testSourceDirectory", "additionalClasspathElements", "argLine",
             "basedir", "childDelegation", "classesDirectory", "classpathDependencyExcludes",
             "classpathDependencyScopeExclude", "debugForkedProcess", "dependenciesToScan",
@@ -107,23 +107,23 @@ public class ExecuteMojoUtil {
     ));
 
     // https://maven.apache.org/surefire/maven-failsafe-plugin/verify-mojo.html
-    private static final ArrayList<String> verifyParams = new ArrayList<>(Arrays.asList(
+    private static final ArrayList<String> VERIFY_PARAMS = new ArrayList<>(Arrays.asList(
             "summaryFile", "basedir", "encoding", "failIfNoTests", "reportsDirectory",
             "skip", "skipExec", "skipITs", "skipTests", "summaryFiles", 
             "testClassesDirectory", "testFailureIgnore"
     ));
 
     // https://maven.apache.org/surefire/maven-surefire-report-plugin/report-only-mojo.html
-    private static final ArrayList<String> reportOnlyParams = new ArrayList<>(Arrays.asList(
+    private static final ArrayList<String> REPORT_ONLY_PARAMS = new ArrayList<>(Arrays.asList(
             "outputName", "showSuccess", "aggregate", "alwaysGenerateSurefireReport",
             "description", "linkXRef", "reportsDirectories", "reportsDirectory",
             "skipSurefireReport", "title", "xrefLocation"
     ));
 
     // https://maven.apache.org/surefire/maven-surefire-report-plugin/failsafe-report-only-mojo.html
-    private static final ArrayList<String> failsafeReportOnlyParams = reportOnlyParams;
+    private static final ArrayList<String> FAILSAFE_REPORT_ONLY_PARAMS = REPORT_ONLY_PARAMS;
 
-    private static final ArrayList<String> libertyCommonParams = new ArrayList<>(Arrays.asList(
+    private static final ArrayList<String> LIBERTY_COMMON_PARAMS = new ArrayList<>(Arrays.asList(
             "installDirectory", "assemblyArchive", "assemblyArtifact", "libertyRuntimeVersion",
             "install", "licenseArtifact", "serverName", "userDirectory", "outputDirectory",
             "assemblyInstallDirectory", "refresh", "skip"
@@ -131,40 +131,40 @@ public class ExecuteMojoUtil {
     // "runtimeArchive", "runtimeArtifact", "runtimeInstallDirectory"
     ));
 
-    private static final ArrayList<String> libertyCommonServerParams = new ArrayList<>(
+    private static final ArrayList<String> LIBERTY_COMMON_SERVER_PARAMS = new ArrayList<>(
             Arrays.asList("serverXmlFile", "configDirectory", "bootstrapProperties", "bootstrapPropertiesFile",
                     "jvmOptions", "jvmOptionsFile", "serverEnvFile"
             // executeMojo can not use alias parameters:
             // "configFile", "serverEnv"
             ));
     
-    private static ArrayList<String> createParams;
+    private static final ArrayList<String> CREATE_PARAMS;
     static {
-        createParams = new ArrayList<>(Arrays.asList(
+        CREATE_PARAMS = new ArrayList<>(Arrays.asList(
                 "template", "libertySettingsFolder", "noPassword"
                 ));
-        createParams.addAll(libertyCommonParams);
-        createParams.addAll(libertyCommonServerParams);
+        CREATE_PARAMS.addAll(LIBERTY_COMMON_PARAMS);
+        CREATE_PARAMS.addAll(LIBERTY_COMMON_SERVER_PARAMS);
     }
     
-    private static ArrayList<String> deployParams;
+    private static final ArrayList<String> DEPLOY_PARAMS;
     static {
-        deployParams = new ArrayList<>(Arrays.asList(
+        DEPLOTY_PARAMS = new ArrayList<>(Arrays.asList(
                 "appsDirectory", "stripVersion", "deployPackages", "timeout", "looseApplication"
                 // executeMojo can not use alias parameters:
                 // "installAppPackages"
                 ));
-        deployParams.addAll(libertyCommonParams);
-        deployParams.addAll(libertyCommonServerParams);
+        DEPLOTY_PARAMS.addAll(LIBERTY_COMMON_PARAMS);
+        DEPLOTY_PARAMS.addAll(LIBERTY_COMMON_SERVER_PARAMS);
     }
     
-    private static ArrayList<String> installFeatureParams;
+    private static final ArrayList<String> INSTALL_FEATURE_PARAMS;
     static {
-        installFeatureParams = new ArrayList<>(Arrays.asList("features"));
-        installFeatureParams.addAll(libertyCommonParams);
+        INSTALL_FEATURE_PARAMS = new ArrayList<>(Arrays.asList("features"));
+        INSTALL_FEATURE_PARAMS.addAll(LIBERTY_COMMON_PARAMS);
     }
 
-    private static final Map<String, String> libertyAliasMap;
+    private static final Map<String, String> LIBERTY_ALIAS_MAP;
     static {
         Map<String, String>tempMap = new HashMap<String, String>();
         tempMap.put("runtimeArtifact", "assemblyArtifact");
@@ -173,7 +173,7 @@ public class ExecuteMojoUtil {
         tempMap.put("configFile", "serverXmlFile");
         tempMap.put("serverEnv", "serverEnvFile");
         tempMap.put("installAppPackages", "deployPackages");
-        libertyAliasMap = Collections.unmodifiableMap(tempMap);
+        LIBERTY_ALIAS_MAP = Collections.unmodifiableMap(tempMap);
     }
 
     /**
@@ -209,7 +209,7 @@ public class ExecuteMojoUtil {
         }
         if (numExec > 1) {
             log.warn(plugin.getArtifactId() + ":" + goal 
-                    + " goal has multiple execution configrations (default to \"" + execId + "\" execution)");
+                    + " goal has multiple execution configurations (default to \"" + execId + "\" execution)");
         }
         
         if (config == null) {
@@ -228,42 +228,42 @@ public class ExecuteMojoUtil {
         switch (executionGoal) {
         case "liberty-maven-plugin:create":
             config = convertLibertyAlias(config);
-            goalConfig = stripConfigElements(config, createParams);
+            goalConfig = stripConfigElements(config, CREATE_PARAMS);
             break;
         case "liberty-maven-plugin:deploy":
             config = convertLibertyAlias(config);
-            goalConfig = stripConfigElements(config, deployParams);
+            goalConfig = stripConfigElements(config, DEPLOTY_PARAMS);
             break;
         case "liberty-maven-plugin:install-feature":
             config = convertLibertyAlias(config);
-            goalConfig = stripConfigElements(config, installFeatureParams);
+            goalConfig = stripConfigElements(config, INSTALL_FEATURE_PARAMS);
             break;
         case "maven-compiler-plugin:compile":
-            goalConfig = stripConfigElements(config, compileParams);
+            goalConfig = stripConfigElements(config, COMPILE_PARAMS);
             break;
         case "maven-compiler-plugin:testCompile":
-            goalConfig = stripConfigElements(config, testCompileParams);
+            goalConfig = stripConfigElements(config, TEST_COMPILE_PARAMS);
             break;
         case "maven-compiler-plugin:resources":
-            goalConfig = stripConfigElements(config, resourcesParams);
+            goalConfig = stripConfigElements(config, RESOURCES_PARAMS);
             break;
         case "maven-compiler-plugin:testResources":
-            goalConfig = stripConfigElements(config, testResourcesParams);
+            goalConfig = stripConfigElements(config, TEST_RESOURCES_PARAMS);
             break;
         case "maven-surefire-plugin:test":
-            goalConfig = stripConfigElements(config, testParams);
+            goalConfig = stripConfigElements(config, TEST_PARAMS);
             break;
         case "maven-failsafe-plugin:integration-test":
-            goalConfig = stripConfigElements(config, integrationTestParams);
+            goalConfig = stripConfigElements(config, INTEGRATION_TEST_PARAMS);
             break;
         case "maven-failsafe-plugin:verify":
-            goalConfig = stripConfigElements(config, verifyParams);
+            goalConfig = stripConfigElements(config, VERIFY_PARAMS);
             break;
         case "maven-surefire-report-plugin:report-only":
-            goalConfig = stripConfigElements(config, reportOnlyParams);
+            goalConfig = stripConfigElements(config, REPORT_ONLY_PARAMS);
             break;
         case "maven-surefire-report-plugin:failsafe-report-only":
-            goalConfig = stripConfigElements(config, failsafeReportOnlyParams);
+            goalConfig = stripConfigElements(config, FAILSAFE_REPORT_ONLY_PARAMS);
             break;
         default:
             goalConfig = config;
@@ -276,17 +276,17 @@ public class ExecuteMojoUtil {
     private static Xpp3Dom convertLibertyAlias(Xpp3Dom config) {
         // convert alias parameter key to actual parameter key
         Xpp3Dom alias;
-        for (String key : libertyAliasMap.keySet()) {
+        for (String key : LIBERTY_ALIAS_MAP.keySet()) {
             alias = config.getChild(key);
             if (alias != null) {
                 if ("runtimeArtifact".contentEquals(key)) {
-                    Xpp3Dom artifact = new Xpp3Dom(libertyAliasMap.get(key));
+                    Xpp3Dom artifact = new Xpp3Dom(LIBERTY_ALIAS_MAP.get(key));
                     for (Xpp3Dom child : alias.getChildren()) {
                         artifact.addChild(child);
                     }
                     config.addChild(artifact);
                 } else {
-                    Element e = (element(name(libertyAliasMap.get(key)), alias.getValue()));
+                    Element e = (element(name(LIBERTY_ALIAS_MAP.get(key)), alias.getValue()));
                     config.addChild(e.toDom());
                 }
             }

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/utils/ExecuteMojoUtil.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/utils/ExecuteMojoUtil.java
@@ -1,0 +1,311 @@
+package io.openliberty.tools.maven.utils;
+
+import static org.twdata.maven.mojoexecutor.MojoExecutor.configuration;
+import static org.twdata.maven.mojoexecutor.MojoExecutor.element;
+import static org.twdata.maven.mojoexecutor.MojoExecutor.name;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.maven.model.Plugin;
+import org.apache.maven.model.PluginExecution;
+import org.apache.maven.plugin.logging.Log;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+import org.twdata.maven.mojoexecutor.MojoExecutor.Element;
+
+public class ExecuteMojoUtil {
+
+    // https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html
+    private static final ArrayList<String> compileParams = new ArrayList<>(Arrays.asList(
+            "annotationProcessorPaths", "annotationProcessors", "compilerArgs", "compilerArgument",
+            "compilerArguments", "compilerId", "compilerReuseStrategy", "compilerVersion", "debug",
+            "debuglevel", "encoding", "excludes", "executable", "failOnError", "failOnWarning",
+            "fileExtensions", "forceJavacCompilerUse", "fork", "generatedSourcesDirectory", 
+            "includes", "jdkToolchain", "maxmem", "meminitial", "multiReleaseOutput", "optimize",
+            "outputFileName", "parameters", "proc", "release", "showDeprecation", "showWarnings",
+            "skipMain", "skipMultiThreadWarning", "source", "staleMillis", "target",
+            "useIncrementalCompilation", "verbose"
+    ));
+
+    // https://maven.apache.org/plugins/maven-compiler-plugin/testCompile-mojo.html
+    private static final ArrayList<String> testCompileParams = new ArrayList<>(Arrays.asList(
+            "annotationProcessorPaths", "annotationProcessors", "compilerArgs", "compilerArgument",
+            "compilerArguments", "compilerId", "compilerReuseStrategy", "compilerVersion", "debug",
+            "debuglevel", "encoding", "executable", "failOnError", "failOnWarning", 
+            "fileExtensions", "forceJavacCompilerUse", "fork", "generatedTestSourcesDirectory", 
+            "jdkToolchain", "maxmem", "meminitial", "optimize", "outputFileName", "parameters",
+            "proc", "release", "showDeprecation", "showWarnings", "skip", "skipMultiThreadWarning",
+            "source", "staleMillis", "target", "testCompilerArgument", "testCompilerArguments",
+            "testExcludes", "testIncludes", "testRelease", "testSource", "testTarget",
+            "useIncrementalCompilation", "verbose"
+    ));
+
+    // https://maven.apache.org/plugins/maven-resources-plugin/resources-mojo.html
+    private static final ArrayList<String> resourcesParams = new ArrayList<>(Arrays.asList(
+            "outputDirectory", "addDefaultExcludes", "delimiters", "encoding", "escapeString",
+            "escapeWindowsPaths", "fileNameFiltering", "filters", "includeEmptyDirs", 
+            "mavenFilteringHints", "nonFilteredFileExtensions", "overwrite", "skip",
+            "supportMultiLineFiltering", "useBuildFilters", "useDefaultDelimiters"
+    ));
+
+    // https://maven.apache.org/plugins/maven-resources-plugin/testResources-mojo.html
+    private static final ArrayList<String> testResourcesParams = new ArrayList<>(Arrays.asList(
+            "outputDirectory", "resources", "addDefaultExcludes", "delimiters", "encoding",
+            "escapeString", "escapeWindowsPaths", "fileNameFiltering", "filters",
+            "includeEmptyDirs", "mavenFilteringHints", "nonFilteredFileExtensions",
+            "overwrite", "skip", "supportMultiLineFiltering", "useBuildFilters",
+            "useDefaultDelimiters"
+    ));
+
+    // https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html
+    private static final ArrayList<String> testParams = new ArrayList<>(Arrays.asList(
+            "testSourceDirectory", "additionalClasspathElements", "argLine", "basedir",
+            "childDelegation", "classesDirectory", "classpathDependencyExcludes", 
+            "classpathDependencyScopeExclude", "debugForkedProcess", "dependenciesToScan",
+            "disableXmlReport", "enableAssertions", "encoding", "environmentVariables",
+            "excludedGroups", "excludes", "excludesFile", "failIfNoSpecifiedTests",
+            "failIfNoTests", "forkCount", "forkMode", "forkedProcessExitTimeoutInSeconds",
+            "forkedProcessTimeoutInSeconds", "groups", "includes", "includesFile",
+            "junitArtifactName", "junitPlatformArtifactName", "jvm", "objectFactory",
+            "parallel", "parallelOptimized", "parallelTestsTimeoutForcedInSeconds",
+            "parallelTestsTimeoutInSeconds", "perCoreThreadCount", "printSummary",
+            "properties", "redirectTestOutputToFile", "remoteRepositories", "reportFormat",
+            "reportNameSuffix", "reportsDirectory", "rerunFailingTestsCount", "reuseForks",
+            "runOrder", "shutdown", "skip", "skipAfterFailureCount", "skipExec", 
+            "skipTests", "suiteXmlFiles", "systemProperties", "systemPropertiesFile",
+            "systemPropertyVariables", "tempDir", "test", "testClassesDirectory",
+            "testFailureIgnore", "testNGArtifactName", "threadCount", "threadCountClasses",
+            "threadCountMethods", "threadCountSuites", "trimStackTrace", "useFile",
+            "useManifestOnlyJar", "useModulePath", "useSystemClassLoader",
+            "useUnlimitedThreads", "workingDirectory"
+    ));
+
+    // https://maven.apache.org/surefire/maven-failsafe-plugin/integration-test-mojo.html
+    private static final ArrayList<String> integrationTestParams = new ArrayList<>(Arrays.asList(
+            "summaryFile", "testSourceDirectory", "additionalClasspathElements", "argLine",
+            "basedir", "childDelegation", "classesDirectory", "classpathDependencyExcludes",
+            "classpathDependencyScopeExclude", "debugForkedProcess", "dependenciesToScan",
+            "disableXmlReport", "enableAssertions", "encoding", "environmentVariables",
+            "excludedGroups", "excludes", "excludesFile", "failIfNoSpecifiedTests",
+            "failIfNoTests", "forkCount", "forkMode", "forkedProcessExitTimeoutInSeconds",
+            "forkedProcessTimeoutInSeconds", "groups", "includes", "includesFile",
+            "junitArtifactName", "junitPlatformArtifactName", "jvm", "objectFactory",
+            "parallel", "parallelOptimized", "parallelTestsTimeoutForcedInSeconds", 
+            "parallelTestsTimeoutInSeconds", "perCoreThreadCount", "printSummary",
+            "properties", "redirectTestOutputToFile", "remoteRepositories", "reportFormat",
+            "reportNameSuffix", "reportsDirectory", "rerunFailingTestsCount", "reuseForks",
+            "runOrder", "shutdown", "skip", "skipAfterFailureCount", "skipExec", "skipITs",
+            "skipTests", "suiteXmlFiles", "systemProperties", "systemPropertiesFile",
+            "systemPropertyVariables", "tempDir", "test", "testClassesDirectory",
+            "testNGArtifactName", "threadCount", "threadCountClasses", "threadCountMethods",
+            "threadCountSuites", "trimStackTrace", "useFile", "useManifestOnlyJar",
+            "useModulePath", "useSystemClassLoader", "useUnlimitedThreads", "workingDirectory"
+    ));
+
+    // https://maven.apache.org/surefire/maven-failsafe-plugin/verify-mojo.html
+    private static final ArrayList<String> verifyParams = new ArrayList<>(Arrays.asList(
+            "summaryFile", "basedir", "encoding", "failIfNoTests", "reportsDirectory",
+            "skip", "skipExec", "skipITs", "skipTests", "summaryFiles", 
+            "testClassesDirectory", "testFailureIgnore"
+    ));
+
+    // https://maven.apache.org/surefire/maven-surefire-report-plugin/report-only-mojo.html
+    private static final ArrayList<String> reportOnlyParams = new ArrayList<>(Arrays.asList(
+            "outputName", "showSuccess", "aggregate", "alwaysGenerateSurefireReport",
+            "description", "linkXRef", "reportsDirectories", "reportsDirectory",
+            "skipSurefireReport", "title", "xrefLocation"
+    ));
+
+    // https://maven.apache.org/surefire/maven-surefire-report-plugin/failsafe-report-only-mojo.html
+    private static final ArrayList<String> failsafeReportOnlyParams = reportOnlyParams;
+
+    private static final ArrayList<String> libertyCommonParams = new ArrayList<>(Arrays.asList(
+            "installDirectory", "assemblyArchive", "assemblyArtifact", "libertyRuntimeVersion",
+            "install", "licenseArtifact", "serverName", "userDirectory", "outputDirectory",
+            "assemblyInstallDirectory", "refresh", "skip"
+    // executeMojo can not use alias parameters:
+    // "runtimeArchive", "runtimeArtifact", "runtimeInstallDirectory"
+    ));
+
+    private static final ArrayList<String> libertyCommonServerParams = new ArrayList<>(
+            Arrays.asList("serverXmlFile", "configDirectory", "bootstrapProperties", "bootstrapPropertiesFile",
+                    "jvmOptions", "jvmOptionsFile", "serverEnvFile"
+            // executeMojo can not use alias parameters:
+            // "configFile", "serverEnv"
+            ));
+    
+    private static ArrayList<String> createParams;
+    static {
+        createParams = new ArrayList<>(Arrays.asList(
+                "template", "libertySettingsFolder", "noPassword"
+                ));
+        createParams.addAll(libertyCommonParams);
+        createParams.addAll(libertyCommonServerParams);
+    }
+    
+    private static ArrayList<String> deployParams;
+    static {
+        deployParams = new ArrayList<>(Arrays.asList(
+                "appsDirectory", "stripVersion", "deployPackages", "timeout", "looseApplication"
+                // executeMojo can not use alias parameters:
+                // "installAppPackages"
+                ));
+        deployParams.addAll(libertyCommonParams);
+        deployParams.addAll(libertyCommonServerParams);
+    }
+    
+    private static ArrayList<String> installFeatureParams;
+    static {
+        installFeatureParams = new ArrayList<>(Arrays.asList("features"));
+        installFeatureParams.addAll(libertyCommonParams);
+    }
+
+    private static final Map<String, String> libertyAliasMap;
+    static {
+        Map<String, String>tempMap = new HashMap<String, String>();
+        tempMap.put("runtimeArtifact", "assemblyArtifact");
+        tempMap.put("runtimeArchive", "assemblyArchive");
+        tempMap.put("runtimeInstallDirectory", "assemblyInstallDirectory");
+        tempMap.put("configFile", "serverXmlFile");
+        tempMap.put("serverEnv", "serverEnvFile");
+        tempMap.put("installAppPackages", "deployPackages");
+        libertyAliasMap = Collections.unmodifiableMap(tempMap);
+    }
+
+    /**
+     * Given the Plugin get the goal execution configuration.
+     *
+     * @param plugin
+     * @param goal
+     * @return configuration for the plugin execution goal
+     */
+    public static Xpp3Dom getPluginGoalConfig(Plugin plugin, String goal, Log log) {
+        Xpp3Dom config = null;
+        String execId = "default";
+        int numExec = 0;
+
+        List<PluginExecution> executions = plugin.getExecutions();
+        if (executions != null) {
+            for (PluginExecution e : executions) {
+                if (e.getGoals() != null && e.getGoals().contains(goal)) {
+                    if (numExec == 0) {
+                        // execution configuration is already merged with the common plugin
+                        // configuration
+                        config = (Xpp3Dom) e.getConfiguration();
+                        execId = e.getId();
+                    }
+                    numExec++;
+                }
+            }
+            if (config == null) {
+                config = (Xpp3Dom) plugin.getConfiguration();
+            }
+        } else {
+            config = (Xpp3Dom) plugin.getConfiguration();
+        }
+        if (numExec > 1) {
+            log.warn(plugin.getArtifactId() + ":" + goal 
+                    + " goal has multiple execution configrations (default to \"" + execId + "\" execution)");
+        }
+        
+        if (config == null) {
+            config = configuration();
+        } else {
+            config = Xpp3Dom.mergeXpp3Dom(configuration(), config);
+            config = validateConfiguration(plugin, goal, config, log);
+        }
+        log.debug(plugin.getArtifactId() + ":" + goal + " configuration\n" + config);
+        return config;
+    }
+    
+    private static Xpp3Dom validateConfiguration(Plugin plugin, String goal, Xpp3Dom config, Log log) {
+        Xpp3Dom goalConfig;
+        String executionGoal = plugin.getArtifactId() + ":" + goal;
+        switch (executionGoal) {
+        case "liberty-maven-plugin:create":
+            config = convertLibertyAlias(config);
+            goalConfig = stripConfigElements(config, createParams);
+            break;
+        case "liberty-maven-plugin:deploy":
+            config = convertLibertyAlias(config);
+            goalConfig = stripConfigElements(config, deployParams);
+            break;
+        case "liberty-maven-plugin:install-feature":
+            config = convertLibertyAlias(config);
+            goalConfig = stripConfigElements(config, installFeatureParams);
+            break;
+        case "maven-compiler-plugin:compile":
+            goalConfig = stripConfigElements(config, compileParams);
+            break;
+        case "maven-compiler-plugin:testCompile":
+            goalConfig = stripConfigElements(config, testCompileParams);
+            break;
+        case "maven-compiler-plugin:resources":
+            goalConfig = stripConfigElements(config, resourcesParams);
+            break;
+        case "maven-compiler-plugin:testResources":
+            goalConfig = stripConfigElements(config, testResourcesParams);
+            break;
+        case "maven-surefire-plugin:test":
+            goalConfig = stripConfigElements(config, testParams);
+            break;
+        case "maven-failsafe-plugin:integration-test":
+            goalConfig = stripConfigElements(config, integrationTestParams);
+            break;
+        case "maven-failsafe-plugin:verify":
+            goalConfig = stripConfigElements(config, verifyParams);
+            break;
+        case "maven-surefire-report-plugin:report-only":
+            goalConfig = stripConfigElements(config, reportOnlyParams);
+            break;
+        case "maven-surefire-report-plugin:failsafe-report-only":
+            goalConfig = stripConfigElements(config, failsafeReportOnlyParams);
+            break;
+        default:
+            goalConfig = config;
+            log.info("skip execution goal configuration validation for " + executionGoal);
+            break;
+        }
+        return goalConfig;
+    }
+
+    private static Xpp3Dom convertLibertyAlias(Xpp3Dom config) {
+        // convert alias parameter key to actual parameter key
+        Xpp3Dom alias;
+        for (String key : libertyAliasMap.keySet()) {
+            alias = config.getChild(key);
+            if (alias != null) {
+                if ("runtimeArtifact".contentEquals(key)) {
+                    Xpp3Dom artifact = new Xpp3Dom(libertyAliasMap.get(key));
+                    for (Xpp3Dom child : alias.getChildren()) {
+                        artifact.addChild(child);
+                    }
+                    config.addChild(artifact);
+                } else {
+                    Element e = (element(name(libertyAliasMap.get(key)), alias.getValue()));
+                    config.addChild(e.toDom());
+                }
+            }
+        }
+        return config;
+    }
+
+    private static Xpp3Dom stripConfigElements(Xpp3Dom config, ArrayList<String> goalParams) {
+        // strip non applicable parameters
+        List<Integer> removeChildren = new ArrayList<Integer>();
+        for (int i=0; i<config.getChildCount(); i++) {
+            if (!goalParams.contains(config.getChild(i).getName().trim())) {
+                removeChildren.add(i);
+            }
+        }
+        Collections.reverse(removeChildren);
+        for (int child : removeChildren) {
+            config.removeChild(child);
+        }
+        return config;
+    }
+}

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/utils/ExecuteMojoUtil.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/utils/ExecuteMojoUtil.java
@@ -149,13 +149,13 @@ public class ExecuteMojoUtil {
     
     private static final ArrayList<String> DEPLOY_PARAMS;
     static {
-        DEPLOTY_PARAMS = new ArrayList<>(Arrays.asList(
+        DEPLOY_PARAMS = new ArrayList<>(Arrays.asList(
                 "appsDirectory", "stripVersion", "deployPackages", "timeout", "looseApplication"
                 // executeMojo can not use alias parameters:
                 // "installAppPackages"
                 ));
-        DEPLOTY_PARAMS.addAll(LIBERTY_COMMON_PARAMS);
-        DEPLOTY_PARAMS.addAll(LIBERTY_COMMON_SERVER_PARAMS);
+        DEPLOY_PARAMS.addAll(LIBERTY_COMMON_PARAMS);
+        DEPLOY_PARAMS.addAll(LIBERTY_COMMON_SERVER_PARAMS);
     }
     
     private static final ArrayList<String> INSTALL_FEATURE_PARAMS;

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/utils/ExecuteMojoUtil.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/utils/ExecuteMojoUtil.java
@@ -232,7 +232,7 @@ public class ExecuteMojoUtil {
             break;
         case "liberty-maven-plugin:deploy":
             config = convertLibertyAlias(config);
-            goalConfig = stripConfigElements(config, DEPLOTY_PARAMS);
+            goalConfig = stripConfigElements(config, DEPLOY_PARAMS);
             break;
         case "liberty-maven-plugin:install-feature":
             config = convertLibertyAlias(config);

